src/keygen/participant.rs

```diff
@@ -367,34 +426,36 @@ impl KeygenParticipant {
         // Check if we've received all the decommits
         let r2_done = self
             .local_storage
-            .contains_for_all_ids::<storage::Decommit>(&self.other_participant_ids);
+            .contains_for_all_ids::<storage::Decommit>(&self.all_participants());
```

Comment:
question: Does this not have the same slightly async behavior issue as above? Is that what the check in `gen_round_two_msgs` does?

Reply:
This one is fine because we add `Decommit` to local storage at the same time we add `Commit`, and we check for `Commit` at the top of the method, so if that check passes, then `Decommit` will exist for the local participant.

Reply:
Nice, thanks.